### PR TITLE
171 simulator   e2e frontend tests for contract examples

### PR DESCRIPTION
**Pull request by bradleySuira**
This includes the first version of e2e test with setup and first test
*How to test:*
- We should have all containers running  except the front end container
- if we run the local frontend, to update the contracts examples with latest please run `npm run cp-examples`
- run `nvm use && npm ci`
- run `npm run test:e2e`

### CODE REVIEW

**Comment by bradleySuira** [comment]
Screenshot 2024-05-22 at 5 06 29 PM
Updated with two examples

**Comment by cristiam86** [comment]
@bradleySuira please pull from main

**Comment by bradleySuira** [comment]
it is up to date from main @cristiam86

File: frontend/test/e2e/ContractExampleStorage.spec.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  it('should open Storage example contract', async () => {
+    await contractsPage.navigate()
+    await contractsPage.waitUntilVisible()
+    await contractsPage.skipTutorial()
+    await contractsPage.openContract('storage.gpy')
+    const tabs = await driver.findElements(By.xpath("//div[contains(@class, 'contract-item')]"))
+    expect(tabs.length, 'Number of tabs should be 2').equal(2)
+  })
+
+  it('should open Run debug page and set constructor arguments for WizardOfCoin', async () => {
```
Comment (cristiam86) [inline frontend/test/e2e/ContractExampleStorage.spec.ts:31]: @bradleySuira wrong contract name

File: frontend/test/e2e/ContractExampleStorage.spec.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    )
+    expect(stateResult, 'get_storage result should be visible').not.null
+
+    const stateResultText = await driver
+      .wait(until.elementTextContains(stateResult, 'Test initial storage'), 5000)
+      .getText()
+    expect(stateResultText, 'get_storage result should be Test initial storage').be.equal('Test initial storage')
+  })
+
+  it('should call ask_for_coin() method', async () => {
```
Comment (cristiam86) [inline frontend/test/e2e/ContractExampleStorage.spec.ts:110]: @bradleySuira wrong method

**Review by cristiam86 — CHANGES_REQUESTED** [review]
small stuff

File: frontend/test/e2e/ContractExampleStorage.spec.ts (hunk 1 above)
Comment (bradleySuira) [reply frontend/test/e2e/ContractExampleStorage.spec.ts]: fixed

File: frontend/test/e2e/ContractExampleStorage.spec.ts (hunk 2 above)
Comment (bradleySuira) [reply frontend/test/e2e/ContractExampleStorage.spec.ts]: fixed

**Review by cristiam86 — APPROVED** [review]

**Pull request merged**